examples/generative/corrdiff/datasets/hrrrmini.py

```diff
@@ -0,0 +1,198 @@
+        return np.full(self.img_shape, np.nan)
+
+    def latitude(self) -> np.ndarray:
+        """Get latitude values from the dataset."""
+        return np.full(self.img_shape, np.nan)
+
+    def input_channels(self) -> List[ChannelMetadata]:
+        """Metadata for the input channels. A list of ChannelMetadata, one for each channel"""
+        inputs = [ChannelMetadata(name=v) for v in self.input_variables]
+        invariants = [
+            ChannelMetadata(name=v, auxiliary=True) for v in self.invariant_variables
+        ]
+        return inputs + invariants
+
+    def output_channels(self) -> List[ChannelMetadata]:
+        """Metadata for the output channels. A list of ChannelMetadata, one for each channel"""
+        return [ChannelMetadata(name=v) for v in self.output_variables]
+
+    def time(self) -> List:
+        """Get time values from the dataset."""
+        datetimes = (
+            datetime.datetime.utcfromtimestamp(t.tolist() / 1e9) for t in self.times
+        )
+        return [_convert_datetime_to_cftime(t) for t in datetimes]
+
+    def image_shape(self) -> Tuple[int, int]:
+        """Get the (height, width) of the data (same for input and output)."""
+        return self.img_shape
+
+    def normalize_input(self, x: np.ndarray) -> np.ndarray:
+        """Convert input from physical units to normalized data."""
+        return (x - self.input_mean) / self.input_std
+
+    def denormalize_input(self, x: np.ndarray) -> np.ndarray:
+        """Convert input from normalized data to physical units."""
+        return x * self.input_std + self.input_mean
+
+    def normalize_output(self, x: np.ndarray) -> np.ndarray:
+        """Convert output from physical units to normalized data."""
+        return (x - self.output_mean) / self.output_std
+
+    def denormalize_output(self, x: np.ndarray) -> np.ndarray:
+        """Convert output from normalized data to physical units."""
+        return x * self.output_std + self.output_mean
+
+    def upsample(self, x):
+        """Extend x around edges with linear extrapolation."""
+        y_shape = (
+            x.shape[0],
+            x.shape[1] * self.upsample_factor,
+            x.shape[2] * self.upsample_factor,
+        )
+        y = np.empty(y_shape, dtype=np.float32)
+        _zoom_extrapolate(x, y, self.upsample_factor)
+        return y
+
+
+def _load_dataset(data_path, group, variables=None, stack_axis=1):
+    with xr.open_dataset(data_path, group=group) as ds:
+        if variables is None:
+            variables = list(ds.keys())
+        data = np.stack([ds[v] for v in variables], axis=stack_axis)
+    return (data, variables)
+
+
+def _load_stats(stats, variables, group):
+    mean = np.array([stats[group][v]["mean"] for v in variables])[:, None, None].astype(
+        np.float32
+    )
+    std = np.array([stats[group][v]["std"] for v in variables])[:, None, None].astype(
+        np.float32
+    )
+    return (mean, std)
+
+
+def _convert_datetime_to_cftime(
```

Comment:
The same utility exists in `modulus,generative.utils`. Please import from there.

Reply:
Done in https://github.com/NVIDIA/modulus/pull/636/commits/3df9d26f94ec5d13075e1bd6b1f919a9b07b114c.